### PR TITLE
Creating Bankrupt Officer Search API skeleton app

### PR DESCRIPTION
**Pull request by YiannisPapastergiou**
Part of Jira Ticket: [BI-6534](https://companieshouse.atlassian.net/browse/BI-6534)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,28 @@
-# bankrupt-officer-search-api
-API for searching for bankrupt officers
+<!-- FIXME update API name -->
```
Comment (toriwilliamsch) [inline README.md:1]: Can you remove the comment please?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,28 @@
...
+# Companies House Bankrupt Officer Search API
+<!-- FIXME update API name -->
```
Comment (toriwilliamsch) [inline README.md:3]: Can you remove the comment please?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,28 @@
...
+* [Git][3]
+
+### Getting Started
+1. Run `make` to build
+2. Run `./start.sh` to run
+
+### Environment Variables
+Name | Description | Mandatory | Location
+--- | --- | --- | ---
+BANKRUPT_OFFICER_SEARCH_API_PORT | Port this application runs on when deployed. | ✓ | start.sh
```
Comment (toriwilliamsch) [inline README.md:19]: This is only relevant for Vagrant, you can remove this from the README

Comment (YiannisPapastergiou) [reply README.md:19]: Should I remove this or specify that it is Vagrant Environment Variables?

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>uk.gov.companieshouse</groupId>
+		<artifactId>companies-house-parent</artifactId>
+		<version>1.2.0</version>
+	</parent>
+	<artifactId>bankrupt-officer-search-api</artifactId>
+	<version>unversioned</version>
+	<name>bankrupt-officer-search-api</name>
+	<description>CH API handling CRUD operations on Bankrupt Officer Search Service</description>
```
Comment (toriwilliamsch) [inline pom.xml:13]: Could you change this description please as it's not doing CRUD operations, it's only performing a read operation at the moment

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,2 @@
+uk.gov.companieshouse.bankruptofficersearch.api.health=/internal/officer-search/scottish-bankrupt-officers/healthcheck
+uk.gov.companieshouse.bankruptofficersearch.api.home=/internal/officer-search/scottish-bankrupt-officers
```
Comment (toriwilliamsch) [inline src/main/resources/application.properties:2]: Shouldn't need this as it's defined in the `routes.yaml`

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -0,0 +1,2 @@
+uk.gov.companieshouse.bankruptofficersearch.api.health=/internal/officer-search/scottish-bankrupt-officers/healthcheck
```
Comment (toriwilliamsch) [inline src/main/resources/application.properties:1]: This could be `healthcheck` instead of the full route

File: pom.xml [hunk 7]
```diff
@@ -0,0 +1,159 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>uk.gov.companieshouse</groupId>
+		<artifactId>companies-house-parent</artifactId>
+		<version>1.2.0</version>
```
Comment (jsimons1989) [inline pom.xml:8]: Why isn't this on the latests version? 1.3.0 has Sonar fixes.

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+	</parent>
+	<artifactId>bankrupt-officer-search-api</artifactId>
+	<version>unversioned</version>
+	<name>bankrupt-officer-search-api</name>
+	<description>CH API for Bankrupt Officer Search Service</description>
+
+	<properties>
+		<java.version>1.8</java.version>
+		<org.mapstruct.version>1.3.0.Final</org.mapstruct.version>
```
Comment (jsimons1989) [inline pom.xml:17]: Please remove unused dependencies

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+		<dependencies>
+			<dependency>
+				<groupId>org.springframework.boot</groupId>
+				<artifactId>spring-boot-dependencies</artifactId>
+				<version>${spring-boot-dependencies.version}</version>
+				<type>pom</type>
+				<scope>import</scope>
+			</dependency>
+		</dependencies>
+	</dependencyManagement>
```
Comment (jsimons1989) [inline pom.xml:39]: What is this needed for?

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+					<annotationProcessorPaths>
+						<path>
+							<groupId>org.mapstruct</groupId>
+							<artifactId>mapstruct-processor</artifactId>
+							<version>${org.mapstruct.version}</version>
+						</path>
+						<!-- other annotation processors -->
+					</annotationProcessorPaths>
+				</configuration>
+			</plugin>
```
Comment (jsimons1989) [inline pom.xml:139]: As we're not using mapstruct can this be simplified please?
```
<plugin>
     <groupId>org.apache.maven.plugins</groupId>
     <artifactId>maven-compiler-plugin</artifactId>
     <version>${maven-compiler-plugin.version}</version>
     <configuration>
         <source>${java.version}</source>
         <target>${java.version}</target>
    </configuration>
</plugin>
```

File: src/test/java/uk/gov/companieshouse/bankruptofficersearch/api/controller/HealthcheckControllerIntegrationTest.java [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+@AutoConfigureMockMvc
+@SpringBootTest
+class HealthcheckControllerIntegrationTest {
+    @InjectMocks
+    private HealthcheckController controller;
+
+    @Test
+    @DisplayName("Successfully returns health status")
+    public void returnHealthStatusSuccessfully() throws Exception {
+        final ResponseEntity<Void> response = controller.getHealthCheck();
```
Comment (jsimons1989) [inline src/test/java/uk/gov/companieshouse/bankruptofficersearch/api/controller/HealthcheckControllerIntegrationTest.java:22]: This is calling the controller directly. It's a unit test not an integration test. `@AutoConfigureMockMvc` and `@SpringBootTest` aren't required.

File: src/test/java/uk/gov/companieshouse/bankruptofficersearch/api/controller/HealthcheckControllerTest.java [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    private HealthcheckController controllerUnderTest;
+
+    @Test
+    @DisplayName("Health check confirms health with HTTP 200")
+    public void applicationHealthcheckRunsSuccessfully(){
+        // When the health endpoint is polled
+        final ResponseEntity<Void> response = controllerUnderTest.getHealthCheck();
+
+        // Then the response is HTTP 200
+        assertThat(response.getStatusCode(), is(HttpStatus.OK));
```
Comment (jsimons1989) [inline src/test/java/uk/gov/companieshouse/bankruptofficersearch/api/controller/HealthcheckControllerTest.java:27]: The comments aren't needed

File: pom.xml (hunk 8 above)
Comment (YiannisPapastergiou) [reply pom.xml:17]: mapstruct is removed

File: pom.xml (hunk 9 above)
Comment (YiannisPapastergiou) [reply pom.xml:39]: in most recent Java API repos that have been created in CH we use dependency management. I have no good explanation on why this is needed for. I follow examples of more than 5 repos I checked.

File: pom.xml (hunk 10 above)
Comment (YiannisPapastergiou) [reply pom.xml:139]: plugin will be cleaned up after the mapstruct property removal

File: src/test/java/uk/gov/companieshouse/bankruptofficersearch/api/controller/HealthcheckControllerIntegrationTest.java (hunk 11 above)
Comment (YiannisPapastergiou) [reply src/test/java/uk/gov/companieshouse/bankruptofficersearch/api/controller/HealthcheckControllerIntegrationTest.java:22]: will clean up the tests

File: src/test/java/uk/gov/companieshouse/bankruptofficersearch/api/controller/HealthcheckControllerTest.java (hunk 12 above)
Comment (YiannisPapastergiou) [reply src/test/java/uk/gov/companieshouse/bankruptofficersearch/api/controller/HealthcheckControllerTest.java:27]: will remove the comments

File: pom.xml (hunk 7 above)
Comment (YiannisPapastergiou) [reply pom.xml:8]: will update the parent version

**Review by toriwilliamsch — APPROVED** [review]

**Pull request merged**